functions/services/cost_data_service.py

```
@@ -0,0 +1,1442 @@
+    """
+    try:
+        # Import firebase_admin here to allow graceful fallback in tests
+        from firebase_admin import firestore
+
+        db = firestore.client()
+        doc_ref = db.collection("costData").document("locationFactors").collection(zip_code).document("data")
+        doc = doc_ref.get()
+
+        if doc.exists:
+            return doc.to_dict()
+        return None
```

Ruff (0.14.8)

Consider moving this statement to an `else` block

(TRY300)